cli/src/cli_util.rs

```diff
@@ -2555,6 +2560,68 @@ pub fn print_conflicted_paths(
     Ok(())
 }
 
+pub fn print_snapshot_stats(
+    ui: &Ui,
+    stats: &SnapshotStats,
+    path_converter: &RepoPathUiConverter,
+) -> io::Result<()> {
+    // It might make sense to add files excluded by snapshot.auto-track to the
+    // untracked_paths, but they shouldn't be warned every time we do snapshot.
+    // These paths will have to be printed by "jj status" instead.
+    if !stats.untracked_paths.is_empty() {
+        writeln!(ui.warning_default(), "Refused to snapshot some files:")?;
+        let mut formatter = ui.stderr_formatter();
+        for (path, reason) in &stats.untracked_paths {
+            let ui_path = path_converter.format_file_path(path);
+            let message = match reason {
+                UntrackedReason::FileTooLarge { size, max_size } => {
+                    // if the size difference is < 1KiB, then show exact bytes.
+                    // otherwise, show in human-readable form; this avoids weird cases
+                    // where a file is 400 bytes too large but the error says something
+                    // like '1.0MiB, maximum size allowed is ~1.0MiB'
+                    let size_diff = size - max_size;
+                    if size_diff <= 1024 {
```

Comment:
Could we compare the humanized form instead? That seems more direct (and it better handles unlikely cases like max_size being 1KiB or 1GiB)

Reply:
Do you mean something like `HumanByteSize(size).to_string() == HumanByteSize(max_size).to_string()`? It will work, but I don't know if user want to see size_diff for GiB case.

Reply:
Yes, that's what I mean. We could could humanize the diff too, perhaps? So something like "42 KiB too large; the maximum size allowed is 1073741824 bytes (1GiB)".

Reply:
I simply removed the condition to print diff + exact sizes. New output is something like
```
Warning: Refused to snapshot some files:
  file.img: 100.0MiB (104857600 bytes); the maximum size allowed is 1.0MiB (1048576 bytes)
```

PTAL.

Reply:
LGTM, thanks